sys/net/application_layer/dhcpv6/client.c

```diff
@@ -484,7 +485,8 @@ static void _parse_advertise(uint8_t *adv, size_t len)
                     uint32_t ia_id = byteorder_ntohl(ia_pd->ia_id);
                     size_t ia_pd_len = byteorder_ntohs(ia_pd->len);
                     size_t ia_pd_orig_len = ia_pd_len;
-
+                    ia_pd_len -= sizeof(dhcpv6_opt_ia_pd_t);
+                    ia_pd_len += sizeof(dhcpv6_opt_t);
```

Comment:
It's a matter of personal taste, but I would put this put this in one line at the declaration of `ia_pd_len` above in just one line:

```C
size_t ia_pd_len = byteorder_ntohs(ia_pd->len) - 
                   (sizeof(dhcpv6_opt_ia_pd_t) - sizeof(dhcpv6_opt_t));
```

It *shouldn't* be much difference, but this might also help the optimizer identifying that you are just subtracting the difference of two constants so it is better able to deal with the operation.

Reply:
OK, with small correction. It has to be
```c
size_t ia_pd_len = byteorder_ntohs(ia_pd->len) - 
                   sizeof(dhcpv6_opt_ia_pd_t + sizeof(dhcpv6_opt_t);
```
since the length in the option is without the option header.

Reply:
May I squash it directly?

Reply:
> since the length in the option is without the option header.

Wait... now I'm confused... `dhcpv6_opt_ia_pd_t` includes the fields of the normal `dhcpv6_opt_t`. And if the length is *without* the option header why do you subtract it?